### PR TITLE
[ADP-3306] Errors refactoring

### PR DESCRIPTION
**Pull request by paweljakubas**

### CODE REVIEW

File: lib/integration/scenarios/Test/Integration/Scenario/API/Byron/Transactions.hs [hunk 1]
```diff
@@ -763,19 +765,23 @@ spec = describe "BYRON_TRANSACTIONS" $ do
...
         \ctx -> runResourceT @IO $ do
             w <- emptyRandomWallet ctx
-            let startTime = "2009-09-09T09:09:09Z"
-            let endTime = "2001-01-01T01:01:01Z"
+            let startTime' =
+                    parseTimeOrError False defaultTimeLocale "%Y-%m-%dT%H:%M:%SZ"
+                    "2009-09-09T09:09:09Z"
+            let endTime' =
+                    parseTimeOrError False defaultTimeLocale "%Y-%m-%dT%H:%M:%SZ"
+                    "2001-01-01T01:01:01Z"
```
Comment (Anviking) [inline lib/integration/scenarios/Test/Integration/Scenario/API/Byron/Transactions.hs]: Can we not reuse `fromText @Iso8601Time` here? 🤔

File: lib/integration/framework/Test/Integration/Framework/TestData.hs [hunk 2]
```diff
@@ -320,15 +318,6 @@ errMsg400WalletIdEncoding :: String
...
     "wallet id should be a hex-encoded string of 40 characters"
 
-errMsg400StartTimeLaterThanEndTime :: String -> String -> String
-errMsg400StartTimeLaterThanEndTime startTime endTime = mconcat
-    [ "The specified start time '"
-    , startTime
-    , "' is later than the specified end time '"
-    , endTime
-    , "'."
-    ]
```
Comment (Anviking) [inline lib/integration/framework/Test/Integration/Framework/TestData.hs:330]: 👍

File: lib/integration/scenarios/Test/Integration/Scenario/API/Byron/Transactions.hs (hunk 1 above)
Comment (Anviking) [reply lib/integration/scenarios/Test/Integration/Scenario/API/Byron/Transactions.hs]: This seems to work:
```haskell
            let Right (Iso8601Time startTime') = fromText "2009-09-09T09:09:09Z"
            let Right (Iso8601Time endTime') = fromText "2001-01-01T01:01:01Z"
            let link = Link.listTransactions' @'Byron w
                    Nothing
                    (Just $ Iso8601Time startTime')
                    (Just $ Iso8601Time endTime')
                    Nothing
                    Nothing
                    Nothing
            r <- request @([ApiTransaction n]) ctx link Default Empty
            expectResponseCode HTTP.status400 r
            decodeErrorInfo r `shouldBe` StartTimeLaterThanEndTime
                (ApiErrorStartTimeLaterThanEndTime startTime' endTime')
```

**Review by Anviking — APPROVED** [review]
👍 Nice! Only one smaller remark/suggestion https://github.com/cardano-foundation/cardano-wallet/pull/4609/files#r1619416453

File: lib/integration/scenarios/Test/Integration/Scenario/API/Byron/Transactions.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/integration/scenarios/Test/Integration/Scenario/API/Byron/Transactions.hs]: thanks! make sense!

**Pull request merged**